### ISSUE TITLE
[backup][YSQL] Incorrect column-ids in restored YSQL table if original table was altered.

### ISSUE THREAD

**Issue by OlegLoginov** [yugabyte/yugabyte-db#6009]
The problem is described in the issue: https://github.com/yugabyte/yugabyte-db/issues/5958
This issue created for YSQL table. (#5958 - for YCQL only due to different fix for YCQL)

**Comment by OlegLoginov** [comment]
The renaming workaround for YSQL.
It works for columns with the same type.
```
yugabyte=# create table t (a int, b int, c int, h int primary key);
CREATE TABLE
yugabyte=# insert into t (a,b,c,h) values (1,2,3,4);
INSERT 0 1
yugabyte=# insert into t (a,b,c,h) values (5,6,7,8);
INSERT 0 1
yugabyte=# select * from t;
 a | b | c | h 
---+---+---+---
 1 | 2 | 3 | 4
 5 | 6 | 7 | 8
(2 rows)
yugabyte=# alter table t drop a;
ALTER TABLE
yugabyte=# select * from t;
 b | c | h 
---+---+---
 2 | 3 | 4
 6 | 7 | 8
(2 rows)
```
Create the DB backup. Restore it (into DB `yb2` in the example.)
```
$ ./bin/ysqlsh -d yb2
yb2=# select * from t;
 b | c | h 
---+---+---
 1 | 2 | 4
 5 | 6 | 8
(2 rows)
yb2=# alter table t rename b to a;
ALTER TABLE
yb2=# alter table t rename c to b;
ALTER TABLE
yb2=# alter table t add c int;
ALTER TABLE
yb2=# select * from t;
 a | b | h | c 
---+---+---+---
 1 | 2 | 4 | 3
 5 | 6 | 8 | 7
(2 rows)
```

**Comment by OlegLoginov** [comment]
Workaround for different column types in YSQL table.
Source table `yugabyte.t4`:
```
yugabyte=# create table t4 (h int primary key, a int, b float, c text);
CREATE TABLE
yugabyte=# insert into t4 (h,a,b,c) values (1,2,3.14,'qqq');
INSERT 0 1
yugabyte=# insert into t4 (h,a,b,c) values (4,5,6.14,'www');
INSERT 0 1
yugabyte=# select * from t4;
 h | a |  b   |  c  
---+---+------+-----
 1 | 2 | 3.14 | qqq
 4 | 5 | 6.14 | www
(2 rows)
yugabyte=# alter table t4 drop a;
ALTER TABLE
yugabyte=# \d t4
                     Table "public.t4"
 Column |       Type       | Collation | Nullable | Default 
--------+------------------+-----------+----------+---------
 h      | integer          |           | not null | 
 b      | double precision |           |          | 
 c      | text             |           |          | 
Indexes:
    "t4_pkey" PRIMARY KEY, lsm (h HASH)
```
Create backup. Restore it into DB `yb7`. Recreate the table columns in correct original order with correct types & names:
```
$ ./bin/ysqlsh -d yb7
yb7=# \d t4
                     Table "public.t4"
 Column |       Type       | Collation | Nullable | Default 
--------+------------------+-----------+----------+---------
 h      | integer          |           | not null | 
 b      | double precision |           |          | 
 c      | text             |           |          | 
Indexes:
    "t4_pkey" PRIMARY KEY, lsm (h HASH)
yb7=# alter table t4 drop b;
ALTER TABLE
yb7=# alter table t4 drop c;
ALTER TABLE
yb7=# alter table t4 add a int;
ALTER TABLE
yb7=# alter table t4 add b float;
ALTER TABLE
yb7=# alter table t4 add c text;
ALTER TABLE
yb7=# \d t4
                     Table "public.t4"
 Column |       Type       | Collation | Nullable | Default 
--------+------------------+-----------+----------+---------
 h      | integer          |           | not null | 
 a      | integer          |           |          | 
 b      | double precision |           |          | 
 c      | text             |           |          | 
Indexes:
    "t4_pkey" PRIMARY KEY, lsm (h HASH)
```
Create backup of the DB `yb7`. Restore it into DB `yb8`:
```
$ ./bin/ysqlsh -d yb8
yb8=# \d t4
                     Table "public.t4"
 Column |       Type       | Collation | Nullable | Default 
--------+------------------+-----------+----------+---------
 h      | integer          |           | not null | 
 a      | integer          |           |          | 
 b      | double precision |           |          | 
 c      | text             |           |          | 
Indexes:
    "t4_pkey" PRIMARY KEY, lsm (h HASH)
yb8=# select * from t4;
 h | a |  b   |  c  
---+---+------+-----
 1 | 2 | 3.14 | qqq
 4 | 5 | 6.14 | www
(2 rows)
```

**Comment by OlegLoginov** [comment]
Fixed by the commit above.

**Comment by OlegLoginov** [comment]
NOTE: the fix added a schema check on the backup-restore step, which can fail due to another issue.
See: https://github.com/yugabyte/yugabyte-db/issues/6886
(Fix - D10365.)

If your backport this fix, do not forget to backport fix for #6886 too.
